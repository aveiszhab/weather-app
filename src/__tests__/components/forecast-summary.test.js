import React from 'react';
import { render } from '@testing-library/react';
import ForecastSummary from '../../components/forecast-summary';


describe('ForecastSummary', () => {
    it('renders correctly to match ForecastSummary snapshot', () => {
        const { asFragment } = render(
            <ForecastSummary
                date={123134}
                temperature={11}
                description='mockDescription'
                icon='200'
            />
        );
        
        expect(asFragment()).toMatchSnapshot();
    });

    
    it('renders the correct date, temperature, description and icon props', () => {
        const { getByTestId } = render(
            <ForecastSummary
                date={123134}
                temperature={11}
                description='mockDescription'
                icon='200'
        />
        );

        expect(getByTestId('date-id')).toHaveClass('date');
        expect(getByTestId('temperature-id')).toHaveClass('temperature');
        expect(getByTestId('description-id')).toHaveClass('description');
        expect(getByTestId('icon-id')).toHaveClass('icon');

    });
    
});