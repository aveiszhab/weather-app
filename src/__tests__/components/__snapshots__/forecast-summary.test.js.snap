// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly to match ForecastSummary snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <div
      class="date"
      data-testid="date-id"
    >
      Thu 4th Jan
    </div>
    <div
      class="temperature"
      data-testid="temperature-id"
    >
      11Â°c
    </div>
    <div
      class="description"
      data-testid="description-id"
    >
      mockDescription
    </div>
    <div
      class="icon"
      data-testid="icon-id"
    >
      <i
        class="wi wi-day-storm-showers"
        name="owm"
      />
    </div>
    <button
      value="123134"
    >
      More Details
    </button>
  </div>
</DocumentFragment>
`;
